anta/tests/stun.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""Test functions related to various STUN settings."""
+
+# Mypy does not understand AntaTest.Input typing
+# mypy: disable-error-code=attr-defined
+from __future__ import annotations
+
+from ipaddress import IPv4Address
+from typing import ClassVar
+
+from pydantic import BaseModel
+
+from anta.custom_types import Port
+from anta.models import AntaCommand, AntaTemplate, AntaTest
+from anta.tools import get_failed_logs, get_value
+
+
+class VerifyStunClient(AntaTest):
+    """
+    Verifies the configuration of the STUN client, specifically the IPv4 source address and port.
+
+    Optionally, it can also verify the public address and port.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the STUN client is correctly configured with the specified IPv4 source address, public address, and port.
+    * Failure: The test will fail if the STUN client is not configured or if the IPv4 source address, public address, or port details are incorrect.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.stun:
+      - VerifyStunClient:
+          stun_clients:
+            - source_address: 172.18.3.2
+              public_address: 172.18.3.21
+              port: 4500
+            - source_address: 100.64.3.2
+              public_address: 100.64.3.21
+              port: 4500
```

Comment:
The input does not allow to have a different port for source and public IP - this should be done differently as there could be a NAT in between (which we can or cannot predict)?